### ISSUE TITLE
Support for oAuth2 securityScheme

### ISSUE THREAD

**Issue by kiranpatel11** [specmatic/specmatic#727]
We can experimenting with specmatic to see if it meets our need.

One of the early bottleneck we found is the limited support for the authentication to only bearer and api key.

Error:
```
Specmatic only supports bearer and api key authentication (header, query) security schemes at the moment
```

As the actual verification of the security headers is not so vital for the contract testing, specmatic should handle the unspported securityschemes in more gracefull manner, provided the application/SUT can also be setup accordingly for the contract testing. e.g. Instead of throwing the FATAL error, can it just log the WARNING for the unsupported securitySchemes, and still continue with the contract testing ignoring the security header/query param ? 
can this be an interim graceful workaround ?

**Comment by harikrishnan83** [comment]
Duplicate of #611

**Comment by harikrishnan83** [comment]
@kiranpatel11 Thanks for raising this issue, we are already in the process of implementing OAuth2 support. This will be part of our upcoming releases shortly.

**Comment by kiranpatel11** [comment]
fixed in #725
